OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -135,7 +135,11 @@ IEnumerable<IRenderable> IRenderModifier.ModifyRender(Actor self, WorldRenderer
 
 		void ITick.Tick(Actor self)
 		{
-			if (!IsTraitDisabled)
+			if (IsTraitDisabled)
+			{
+				Uncloak();
```

Comment:
When the trait get disabled and re-enabled we want the cloaking timer to be resetted.

Reply:
This will set the countdown timer to `Info.CloakDelay`, which will override the `InitialDelay` value that we want if the trait starts off disabled (e.g. being enabled on veterancy or from a crate).  It would be better to implement and call this from `TraitDisabled`, which only runs when the trait goes from enabled &rarr; disabled.

Reply:
done